chalice/package.py

```diff
@@ -539,15 +566,290 @@ def _register_cfn_resource_name(self, name):
+            'lambda_function_arn': self._fref(resource.lambda_function)
+        }
+
+        if resource.prefix:
+            bnotify['filter_prefix'] = resource.prefix
+        if resource.suffix:
+            bnotify['filter_suffix'] = resource.suffix
+
+        # we use the bucket name here because we need to aggregate
+        # all the notifications subscribers for a bucket.
+        # Due to cyclic references to buckets created in terraform
+        # we also try to detect and resolve.
+        if '{aws_s3_bucket.' in resource.bucket:
+            bucket_name = resource.bucket.split('.')[1]
+        else:
+            bucket_name = resource.bucket
+        template['resource'].setdefault(
+            'aws_s3_bucket_notification', {}).setdefault(
+                bucket_name + '_notify',
+                {'bucket': resource.bucket}).setdefault(
+                    'lambda_function', []).append(bnotify)
+
+        template['resource'].setdefault('aws_lambda_permission', {})[
+            resource.resource_name] = {
+                'statement_id': resource.resource_name,
+                'action': 'lambda:InvokeFunction',
+                'function_name': resource.lambda_function.function_name,
+                'principal': 's3.amazonaws.com',
+                'source_arn': 'arn:aws:s3:::%s' % resource.bucket
+        }
+
+    def _generate_sqseventsource(self, resource, template):
+        # type: (models.SQSEventSource, Dict[str, Any]) -> None
+        template['resource'].setdefault('aws_lambda_event_source_mapping', {})[
+            resource.resource_name] = {
+                'event_source_arn': self._arnref(
+                    "arn:aws:sqs:%(region)s:%(account_id)s:%(queue)s",
+                    queue=resource.queue),
+                'batch_size': resource.batch_size,
+                'function_name': resource.lambda_function.function_name,
+        }
+
+    def _generate_snslambdasubscription(self, resource, template):
+        # type: (models.SNSLambdaSubscription, Dict[str, Any]) -> None
+
+        if resource.topic.startswith('arn:aws'):
+            topic_arn = resource.topic
+        else:
+            topic_arn = self._arnref(
+                'arn:aws:sns:%(region)s:%(account_id)s:%(topic)s',
+                topic=resource.topic)
+
+        template['resource'].setdefault('aws_sns_topic_subscription', {})[
+            resource.resource_name] = {
+                'topic_arn': topic_arn,
+                'protocol': 'lambda',
+                'endpoint': self._fref(resource.lambda_function)
+        }
+        template['resource'].setdefault('aws_lambda_permission', {})[
+            resource.resource_name] = {
+                'function_name': resource.lambda_function.function_name,
+                'action': 'lambda:InvokeFunction',
+                'principal': 'sns.amazonaws.com',
+                'source_arn': topic_arn
+        }
+
+    def _generate_scheduledevent(self, resource, template):
```

Comment:
while its good to note in the pr, this should be filed as a new issue on the repo so it can be tracked for future work. at least for terraform we could indeed be outputing the deployed.json resources with these resources, and technically that may allow some degree of bidirectional compatibility for deploy/package w/ terraform.

Reply:
given the other pull requests in the queue, and the nature of the code structure creating conflicts, I went ahead and started on this to have the terraform backend be feature complete, so I can context switch to the other features, as incremental on this would naturally conflict with the other queue items.